### PR TITLE
[SPARK-583] New Spark distribution with back ported PRs

### PR DESCRIPTION
**Pull request by ArtRand**
## Back ported the following PRs to `custom-2.2.0-X`:
Sandbox URL: https://github.com/apache/spark/pull/18528 DONE
Ticket renewal mistake: https://github.com/apache/spark/pull/19798 
Checkpointing and Supervise: https://github.com/apache/spark/pull/19374 *added unit test*
https://www.cloudera.com/documentation/enterprise/5-5-x/topics/spark_mllib.html

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -71,7 +71,7 @@ prod-dist: $(SPARK_DIR)
 		else \
 			MESOS_PROFILE=""; \
 		fi; \
-		./dev/make-distribution.sh --tgz "$${MESOS_PROFILE}" "-Phadoop-$(HADOOP_VERSION)" -Psparkr -Phive -Phive-thriftserver -DskipTests; \
+		./dev/make-distribution.sh --tgz "$${MESOS_PROFILE}" "-Phadoop-$(HADOOP_VERSION)" -Pnetlib-lgpl -Psparkr -Phive -Phive-thriftserver -DskipTests; \
```
Comment (susanxhuynh) [inline Makefile]: What is this for?

File: tests/test_hdfs.py [hunk 2]
```diff
@@ -125,3 +128,65 @@ def test_terasort_suite():
                     expected_output="partitions are properly sorted",
                     app_name="/spark",
                     args=teravalidate_args)
+
+
+@pytest.mark.skipif(not utils.hdfs_enabled(), reason='HDFS_ENABLED is false')
+@pytest.mark.sanity
+def test_supervise():
```
Comment (susanxhuynh) [inline tests/test_hdfs.py]: There's already a test with the same name in test_spark.py. Did you want to rename it to something like test_supervise_with_checkpointing?

File: Makefile (hunk 1 above)
Comment (ArtRand) [reply Makefile]: native BLAS

File: tests/test_hdfs.py (hunk 2 above)
Comment (ArtRand) [reply tests/test_hdfs.py]: ah no, I should remove the other test.

**Pull request merged**